packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -402,14 +420,16 @@ function getSelectedItemsFromValue<T extends string | number>(
   const resultingItems: Array<ComboboxOption<T>> = [];
 
   for (const option of options) {
-    const index = valueMap.get(option.value);
-    if (index !== undefined) {
-      resultingItems[index] = option;
-      valueMap.delete(option.value);
-    }
-    if (valueMap.size === 0) {
-      // We found all values
-      break;
+    if (isComboboxOption(option)) {
```

Comment:
You might want to do an early return/continue by doing `!isComboboxOption(option)` instead